agent/discovery/query_fetcher_v2.go

```diff
@@ -91,10 +94,15 @@ func (f *V2DataFetcher) FetchEndpoints(reqContext Context, req *QueryPayload, lo
 	}
 
 	results := make([]*Result, 0, limit)
-	for idx := 0; idx < limit; idx++ {
-		endpoint := serviceEndpoints.Endpoints[idx]
-
-		// TODO (v2-dns): filter based on the port name requested
+	for _, endpoint := range serviceEndpoints.Endpoints[:limit] {
+
+		// First we check the endpoint first to make sure that the requested port is matched from the service.
+		// We error here because we expect all endpoints to have the same ports as the service.
+		ports := getResultPorts(req, endpoint.Ports) //assuming the logic changed in getResultPorts
+		if ports == nil || len(ports) == 0 {
```

Comment:
I think len() will convert nil to zero for you.

Reply:
Good point, will update